### ISSUE TITLE
Erasing Concepts from Diffusion Models

### ISSUE THREAD

**Issue by nmhkahn** [nmhkahn/paper-reading#68]
## Motivation
- SD는 다양한 이미지를 생성하지만 몇몇 concept은 생성되면 안되거나 (NSFW 등), 저작권 이슈가 있을 수 있음
- 따라서 이러한 컨셉을 선택적으로 지우는 연구 필요
- NSFW 필터 등의 후처리 방법은 사용자가 회피할 수 있기 때문에 모델 파라미터 내에서 제거되는 편이 좋으나, SD v2.0처럼 이를 고려해서 다시 학습하는 방법은 현실적이지 않음
- Pre-trained SD를 fine-tune해서 모델 파라미터 내에 undesired 컨셉을 지우는 연구

## Method
<img width="1029" alt="Screenshot 2023-03-28 at 10 02 30 AM" src="https://user-images.githubusercontent.com/16869368/228099965-89fe7666-fe32-4476-a212-d6d67bc80f99.png">

- Classifier-free guidance에 영감을 받아서 모델이 특정 컨셉 $c$에 멀어지도록 유도(steering)
    - CFG는 아래 수식처럼 cond/uncond 시나리오의 생성 결과를 통해 인퍼런스 시에 모델 가이드
    - <img width="300" alt="Screenshot 2023-03-28 at 10 03 22 AM" src="https://user-images.githubusercontent.com/16869368/228100071-2c38e1f0-4b0c-4037-8fd2-5ddc65d26ca3.png">
- Score-based formulation으로 컨셉을 negate 하는 수식을 아래처럼 유도
    - <img width="300" alt="Screenshot 2023-03-28 at 10 08 12 AM" src="https://user-images.githubusercontent.com/16869368/228100607-e6b2371d-10b8-4397-a0c6-5b64b1029461.png">
- CFG와 달리 frozen SD $\theta^\*$와 fine-tuned SD $\theta$ ($\theta^\*$에서 카피)를 가지고 업데이트
    - $x_t$를 뽑고 $\theta$에 한 번, $\theta^\*$에 cond/uncond 두 번 뽑아서 $\epsilon_\theta(.)$를 계산한 뒤 L2 loss
- 이 때, 어떤 컨셉을 지우냐에 따라 fine-tune 할 때 필요한 레이어가 달라짐
    - ESD-x: cross attention 레이어 fine-tune; 아티스트 이름과 같이 specific prompt를 제거할 때 좋음
    - ESD-u: non-cross attention 레이어 fine-tune; NSFW처럼 텍스트에 독립적인 컨셉 제거할 때 좋음

**Comment by nmhkahn** [comment]
## Motivation
- 학습 시 모델 두개를 올려야하기 때문에 조금 비쌈
- 아마 모델 파라미터를 직접적으로 fine-tune하기 때문에 concept erase와 surrounding concept quality와의 trade-off가 있을 가능성이 매우 큼. 이걸 잘 reg 하는 loss 디자인이 필요
- Continual하게 concept erase하는 시나리오에서 잘 working할지? - 이 논문의 motivation 처럼 하려면 엄청 많은 concept을 지우는게 필요한데 이런 경우 모델 파라미터 측면에서 어떤 변화가 일어날지 장담하기 어려움
